### PR TITLE
feature: config option to disable text boxes

### PR DESCRIPTION
**Pull request by FelixAbrahamsson**

### CODE REVIEW

File: src/core/createPolygon.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ const createPolygon = (state) => ({
   coordinates: [],
   visible: true,
   color: getUIColor(state),
+  disableTextBoxes: state?.config?.disableTextBoxes === true ? true : false,
```
Comment (fippli) [inline src/core/createPolygon.js]: isn't this the same as 
```
disableTextBoxes: state?.config?.disableTextBoxes
```

My suggestion is to have the other way around
```
showTextBoxes: state?.config?.showTextBoxes
```

default to `false`

what do you think about that?

File: src/core/drawPolygon.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ const drawPolygon = (canvas, state) => (polygon, index) => {
         );
 
         // Draw index in the corner of the box
-        if (index !== undefined) {
+        if (!state?.config?.disableTextBoxes && index !== undefined) {
```
Comment (fippli) [inline src/core/drawPolygon.js]: Makes sense here also
```
if (state.config.displayTextBoxes) {
  // go ahead and render
}
```

File: src/core/createPolygon.js (hunk 1 above)
Comment (fippli) [reply src/core/createPolygon.js]: ah, `undefined` and `null` mess this up?

Comment (FelixAbrahamsson) [reply src/core/createPolygon.js]: I like that more, it's less confusing, I just thought we should preserve the same defaults :) But maybe that doesn't matter

Comment (FelixAbrahamsson) [reply src/core/createPolygon.js]: I wanted to force it into true/false yes so we dont have to handle undefined/null anywhere else. But maybe that also doesnt matter

Comment (fippli) [reply src/core/createPolygon.js]: This is still alpha/beta so some breaking changes are ok. It's better to find the good solutions at this point. We don't need backwards compability.

**Review by fippli — APPROVED** [review]

**Pull request merged**